externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala

```diff
@@ -115,16 +115,16 @@ class ExecuteStatement(
       while (loop) {
         Thread.sleep(50) // slow the processing down
 
-        val pageSize = Math.min(500, resultMaxRows)
-        val result = executor.snapshotResult(sessionId, resultId, pageSize)
+        val result = executor.snapshotResult(sessionId, resultId, resultMaxRows)
```

Comment:
If I understand correctly, this change fixes a correctness issue, but breaks the original design - retrieve the result in small pages to avoid OOM, am I right?

also cc @link3280 and @bowenliang123

Reply:
In my opinion, if in order to retrieve the result in small pages to avoid OOM,  suggest kyuubi start to check kyuubi.session.engine.flink.max.rows value and provide detailed explanations on the official website. 
Now  val pageSize = Math.min(500, resultMaxRows),  dead value(500) is imperfect.

Reply:
> If I understand correctly, this change fixes a correctness issue, but breaks the original design - retrieve the result in small pages to avoid OOM, am I right?
> 
> also cc @link3280 and @bowenliang123

Not really, the row limit is introduced to avoid OOM. The page size is designed to keep fetching rows until the limit is reached or all records are read, because we may not get enough rows with one fetch (consider Kafka as the source).